proxy/http/HttpSM.cc

```diff
@@ -3378,7 +3378,8 @@ HttpSM::tunnel_handler_cache_write(int event, HttpTunnelConsumer *c)
     } else {
       *status_ptr      = HttpTransact::CACHE_WRITE_COMPLETE;
       c->write_success = true;
-      c->write_vio     = c->vc->do_io(VIO::CLOSE);
+      c->write_vio     = nullptr;
+      c->vc->do_io_close();
```

Comment:
I think it'd be safer to do this in the opposite order - do the clean up then clear the `write_vio`.